### ISSUE TITLE
ww-poreCoV extension

### ISSUE THREAD

**Issue by hoelzer** [replikation/poreCov#275]
I suggest using the poreCov pipeline as the backend for SARS-CoV-2 wastewater lineage deconvolution from nanopore long reads. You already added `freyja` ( #274 #270), which is great as the current community standard. 

However, we are also interested in detecting new stuff, aka "cryptic lineages" or novel mutation profiles. 

To do this, I would like to test/implement two recent approaches:

1) `CONCOMPRA`
- https://github.com/willem-stock/CONCOMPRA
- consensus approach for community profiling with nanopore amplicon sequencing data
- I already tested this on example nanopore data from mixed patient samples (simulating wastewater) and it looked very promising
- should be easy to add as a single new process

2) `Floria`
- https://github.com/bluenote-1577/floria | https://doi.org/10.1093/bioinformatics/btae252
- Strain-level haplotyping for metagenomes with short or long reads.
- I am curious how well this works on a) amplicon data and b) SARS-CoV-2
- input is VCF and mapped reads (BAM): so I think should be also easy to add 

By this, we would get known lineage abundances from `freyja` plus potential new lineages from one or both of the other tools. 

Finally, we could also write a little ww-poreCoV extension paper ;)

**Comment by hoelzer** [comment]
Another interesting method might be VirPool: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-022-05100-3

They illustrated already the advantage that longer amplicons bring w/ their method in particular. 

> is able to use the entire length of reads instead of just the most informative positions, and can also capture haplotype dependencies within a single read. A crucial property of our model is its ability to capture long-range dependencies within reads, which is particularly relevant when coupled with use of long amplicons and nanopore sequencing.

They also tested the method on real data by sequencing a mixture of eight clinical samples using long amplicons (2kb).

(sidenote, attention was brought up and information was shared by Victor! thx!)

**Comment by replikation** [comment]
Yep, it's completely fine to extend Porecov now on the wastewater surveillance. We just need to make sure the normal "user experience" is not convoluted.

**Comment by hoelzer** [comment]
> ad `Florida`:
> 
> You would use the VCF + BAM form ARTIC, right? Just to keep in mind: mixed indels might be tricky. I saw an overlapping x nt deletion and y nt deletion, and neither of both was called by medaka, and/or a frameshift introducing indel was called

Yes, I would like to use the output porecov anyway produces to change as little as possible. 

However, good point. I would live with such issues for now. Deconvoluting lineages from wastewater is anyway wild west :) but of course, important to keep such situations in mind.

**Comment by replikation** [comment]
Not sure if it is possible to implement a subcommand or a second "main.nf" solely for the wastewater part? (thinking samtools subcommand for instance)

**Comment by hoelzer** [comment]
Started working on this in a branch `ww-porecov`. First step was a container for `CONCOMPRA` which was already a bit pain, but made it finally

`rkimf1/concompra:0.0.0--f3c7699`

**Comment by hoelzer** [comment]
The authors of `CONCOMPRA` provide a docker now: `willemstock/concompra:version0.0.2`

However, we need to see if that works w/ nextflow

**Comment by hoelzer** [comment]
This might be even more interesting to add as a process to ww-poreCov:

* https://www.medrxiv.org/content/10.1101/2024.08.27.24312690v1
* Unsupervised detection of SARS-CoV-2 mutations and lineages in Norwegian wastewater samples using long-read sequencing
* https://github.com/garcia-nacho/HERCULES

One big questions is probably: how do they handle the reference used for lineage assignments at the end? And: **can they detect something now/cryptic**

**Comment by hoelzer** [comment]
Looking into `HERCULES`

* they have a pre-build docker container but we might want our own to work w/ nextflow (its maybe a bit cumbersome to use otherwise)

* they provide option for own reference, based on aligned spike genes:

> To update the set of references that HERCULES uses, you need a fasta file with all the Spike-genes of the reference sequences aligned. The names of the sequences inside the fasta file must have the following structure IDXX_LineageY where IDXX is the unique identifier for the sequence and the LineageY is the lineage assigned to the sequence, it can be pangolin lineage or any other nomenclature system (e.g. nextclade clade ID, WHO nomenclature, etc). The fasta file must be stored in a folder that must be mounted inside HERCULES with the following flag. -v Path/to/Folder:/Reference

This might be a good option to build an up-to-date reference based on "representative spike sequences" per lineage (https://github.com/rki-mf1/sc2-mutation-frequency-calculator)

**Comment by hoelzer** [comment]
re-tagged the provided image to `mhoelzer/hercules:0.0.0` bc the provided name (`ghcr.io/garcia-nacho/hercules`) crashes some process in poreCov

**Comment by hoelzer** [comment]
Unfortunately, the container is a pain to use in Nextflow. Here is the Dockerfile and how then the tool is executed (a bash script):

https://github.com/garcia-nacho/HERCULES/blob/master/Dockerfile#L76C19-L76C57

Probably, we need first to re-build the container. I tried to just `docker build` the original Dockerfile but this fails early (on my MacBook).

**Comment by hoelzer** [comment]
Is this maybe interesting for our purpose of detecting smt new? 

https://github.com/zhuangx15/ICAvar
https://www.medrxiv.org/content/10.1101/2024.04.18.24306052v1.full

**Comment by hoelzer** [comment]
Another idea:

devider; https://www.biorxiv.org/content/10.1101/2024.11.05.621838v1

**Comment by hoelzer** [comment]
Another package from colleagues in France (Hughes has contact) where they reimplemented ww tools and making them faster

Also

> ... indeed the haplotype callers should do the task that you need. If you want to make full use of the long reads, you will need to start from the BAM and count the mutation cooccurrences. The tool from El Hacene DJAOUT does that: https://pypi.org/project/VaRaPS/1.0.0

**Comment by hoelzer** [comment]
> Another interesting method might be VirPool: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-022-05100-3
> 
> They illustrated already the advantage that longer amplicons bring w/ their method in particular.
> 
> > is able to use the entire length of reads instead of just the most informative positions, and can also capture haplotype dependencies within a single read. A crucial property of our model is its ability to capture long-range dependencies within reads, which is particularly relevant when coupled with use of long amplicons and nanopore sequencing.
> 
> They also tested the method on real data by sequencing a mixture of eight clinical samples using long amplicons (2kb).
> 
> (sidenote, attention was brought up and information was shared by Victor! thx!)
> 
> The question is, as usually, how up-to-date is the tool?
> 
> https://github.com/fmfi-compbio/virpool
> 
> They also provide scripts to create own profiles: https://github.com/fmfi-compbio/virpool?tab=readme-ov-file#creating-a-custom-variant-profile
> 
> Maybe `covsonar` could also create these profiles easily. Or Ashkans `sc2mfc` tool.

Virpool might be promising... 

> you have the choice between : Virpool (cooccurrence), Frejya (frequencies), and LCS (counts) 

And the tool in the post before has a faster reimplementation of virpool

**Comment by hoelzer** [comment]
Thomas is testing: https://github.com/bluenote-1577/devider

**Comment by hoelzer** [comment]
That looks also promising, VILOCA:

https://academic.oup.com/nargab/article/6/4/lqae152/7912062

![Image](https://github.com/user-attachments/assets/741d286b-065a-4ba0-a70a-9f22ceb3036c)

**Comment by hoelzer** [comment]
it might be really interesting to maybe replace the current Freyja in poreCov with VaRaPS if that even allows more flexibility in reference database M design and access to a method that works better with co-occurrence information.
